amkotlinutil/src/main/java/dk/adaptmobile/amkotlinutil/navigation/BaseView.kt

```diff
@@ -0,0 +1,129 @@
+                        .doOnAndroidMain()
+                        .subscribe(
+                                {
+                                    if (controller.isAttached) {
+                                        handleOutput(it)
+                                    }
+                                },
+                                {
+                                    e(this.javaClass.name, "Error in BaseView output: $it")
+                                }
+                        )
+            }
+
+            override fun postCreateView(controller: Controller, view: View) {
+                super.postCreateView(controller, view)
+                activity?.let { activity ->
+                    onViewBound(view, activity)
+
+                    viewModel.loading
+                            .compose(bindUntilEvent(ControllerEvent.DESTROY))
+                            .doOnAndroidMain()
+                            .subscribe {
+//                                showLoadingBar(it)
+                            }
+
+                    viewModel.error
+                            .compose(bindUntilEvent(ControllerEvent.DESTROY))
+                            .doOnAndroidMain()
+                            .subscribe {
+                                //TODO: Handle error
+                            }
+
+                    val application = activity.application as BaseApplicationController
+                    application.noNetworkSubject
+                            .compose(bindUntilEvent(ControllerEvent.DESTROY))
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .subscribe({ hasNetwork ->
+                                           if (!hasNetwork) {
+                                               d(this.javaClass.name, "NoInternetDialog opens")
+//                                               NavManager.openModally(Routing.NoInternetDialog())
+                                           }
+                                       }, {
+//                                           NavManager.openModally(Routing.NoInternetDialog())
+                                       })
+                }
+            }
+        }
+        addLifecycleListener(listener)
+    }
+
+    protected abstract fun setViewModel(): T
+
+    protected abstract fun inflateView(): Int
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup, savedViewState: Bundle?): View  {
+        this.savedViewState = savedViewState
+        return container.inflate(inflateView(), false)
+    }
+
+    protected abstract fun onViewBound(view: View, activity: Activity)
+
+    protected abstract fun handleOutput(output: T2)
+
+    open fun callback(data: Any?) {}
+
+    override fun onDestroyView(view: View) {
+        super.onDestroyView(view)
+        clearFindViewByIdCache()
+        viewModel.disposeBag.disposeSafe()
+    }
+
```

Comment:
This branch didn't have the ktlint? Maybe run ktlint after it's been merged to master 😉